arelle/plugin/validate/DBA/rules/__init__.py

```diff
@@ -73,12 +73,38 @@ def errorOnRequiredFact(
 )
 
 
-def findFactsWithDimension(
+def errorOnRequiredPositiveFact(
+        modelXbrl: ModelXbrl,
+        facts: set[ModelFact | None],
+        code: str,
+        message: str,
+) -> Iterable[Validation]:
+    """
+    Yields an error if a fact with the given QName is not tagged with a valid date and a non-nil value.
+    :return: Yields validation errors.
+    """
+    errorModelObjects = []
+    if not facts:
+        errorModelObjects.append(modelXbrl.modelDocument)
+    else:
+        for fact in facts:
+            if fact.xValue < 0:
```

Comment:
I think that this will throw a Type error if a fact is tagged without a value such that xValue is None: `TypeError: '<' not supported between instances of 'NoneType' and 'int'`. So I think you also need to add an xValid check. So something like the following:

`if fact.xValid >= VALID and fact.xValue < 0: `

Reply:
I added the check to the file but I don't think that adding a conformance suite test is of any benefit as I can't envision a test that would be testing the validation and not core Arelle.